### DISCUSSION TITLE
v6 : get exact position of object on canvas

### DISCUSSION THREAD

**Discussion by Clemweb** [fabricjs/fabric.js#9838] [Q&A] [answered]
Hi,

Is there anybody to tell me how to get the exact position of an object on the canvas (or the center position) ?

I'm using getBoundingRect or getAbsoluteCoords, but the results are wrong (I calculate in pixels screenshot / photoshop)

I believe something has change with calcViewportBoundaries but I can't understand why...

Thanks

**Comment by ShaMan123** [comment, marked as answer]
Depends if you want the data to be in the viewport/camera or in the scene, meaning does zooming/panning affect your data? If so you are working in the viewport/camera plane

```
const sceneCoords = object.getCoords()
const centerPoint = object.getCenterPoint() === sceneCoords[0].midPointFrom(sceneCoords[2]) // tl.midPointFrom(br)

const viewportCoords = sceneCoords.map(c => fabric.util.sendPointToPlane(c, canvas.viewportTransform))
// center point can be obtained in the same way, using sendPointToPlane, or midPointFrom

const bbox = fabric.util.makeBoundingBoxFromPoint(points)
```
